src/camera.h

```diff
@@ -71,20 +73,25 @@ class camera
             if ((*world)->hit(cur_ray, hit_interval, rec)) {
                 ray scattered;
                 float3 attenuation; //means color
+                float3 color_from_emission = rec.mat_ptr->emitted();
                 if (rec.mat_ptr->scatter(cur_ray, rec, attenuation, scattered, local_rand_state)) {
-                    cur_attenuation *= attenuation;
+                    cur_attenuation *= attenuation + color_from_emission;
                     cur_ray = scattered;
                 }
                 else {
-                    return make_float3(0.0,0.0,0.0);
+                    cur_attenuation *= color_from_emission;
+                    return cur_attenuation;
+                    
+                    // return make_float3(0.0,0.0,0.0);
                 }
             }
             else {
                 //background
-                float3 unit_direction = normalize(cur_ray.direction());
-                float t = 0.5f*(unit_direction.y + 1.0f);
-                float3 c = (1.0f-t)*make_float3(1.0, 1.0, 1.0) + t*make_float3(0.5, 0.7, 1.0);
-                return cur_attenuation * c;
+                return cur_attenuation*background_color;
+                // float3 unit_direction = normalize(cur_ray.direction());
+                // float t = 0.5f*(unit_direction.y + 1.0f);
+                // float3 c = (1.0f-t)*make_float3(1.0, 1.0, 1.0) + t*make_float3(0.5, 0.7, 1.0);
+                // return cur_attenuation * c;

```

Comment:
Same here :)

Reply:
Kinda want to leave it in cus lights to be bright need a lot of samples per pixel this background gives us soft iluminatio much faster.